bundles/org.openhab.core.transform/src/test/java/org/openhab/core/transform/ManagedTransformationConfigurationProviderTest.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.transform;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.openhab.core.common.registry.ProviderChangeListener;
+import org.openhab.core.test.storage.VolatileStorageService;
+
+/**
+ * The {@link ManagedTransformationConfigurationProviderTest} includes tests for the
+ * {@link org.openhab.core.transform.ManagedTransformationConfigurationProvider}
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.WARN)
+@NonNullByDefault
+public class ManagedTransformationConfigurationProviderTest {
+
+    @Mock
+    private @NonNullByDefault({}) ProviderChangeListener<@NonNull TransformationConfiguration> listener;
```

Comment:
You can save one line by moving the `@Mock` annotation. When `Mock` is added as variable name suffix, it will be easier to understand in the code what kind of behavior such objects have. It would ofcourse also be nice to do this in the other new code. :slightly_smiling_face: 

```suggestion
    private @Mock @NonNullByDefault({}) ProviderChangeListener<@NonNull TransformationConfiguration> listenerMock;
```